### PR TITLE
qa_crowbarsetup: Adjust neutron proposals for Cloud5+

### PR DESCRIPTION
**Pull request by toabctl**
**UNTESTED**. And must definitly merged after https://github.com/crowbar/barclamp-neutron/pull/175 and https://github.com/crowbar/barclamp-nova/pull/419 .

Due to changes in barclamp-neutron, new attribute names must be used.
Also enable gre and vlan ml2 type drivers in parallel when using
openvswitch.

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1350,6 +1350,22 @@ function custom_configuration()
             if [ -n "$networkingplugin" ] ; then
                 proposal_set_value neutron default "['attributes']['neutron']['networking_plugin']" "'$networkingplugin'"
             fi
+            # For Cloud5+, proposal attribute names changed
+            if iscloudver 5plus; then
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: Change to `iscloudver 5plus && [[ ! $cloudsource =~ ^M[1-4]+$ ]]`

Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: Argh, no, that's wrong. Hrm, it's really: "if Cloud 5 and later than M4, or Cloud 6+, then"

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1350,6 +1350,22 @@ function custom_configuration()
...
+                if [ "$networkingplugin" = openvswitch ] ; then
+                    proposal_set_value neutron default "['attributes']['neutron']['networking_plugin']" "'ml2'"
+                    proposal_set_value neutron default "['attributes']['neutron']['ml2_type_drivers']" "['gre','vlan']"
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: Shouldn't all this be something we can change with parameters?

Comment (toabctl) [inline scripts/qa_crowbarsetup.sh]: > Shouldn't all this be something we can change with parameters?

Yes. But as it is now, we don't have to change something on the Jenkins side and can get the ml2 things merged as fast as possible. I would like to have new parameters in an extra commit. And then we should test openvswitch with vlan and gre separately. Does this make sense?

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (toabctl) [inline scripts/qa_crowbarsetup.sh]: Are we expecting people using mkcloud together with milestone releases? For other features related to Cloud5, we also use ```iscloudver 5plus``` ...

Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: currently QA still uses M4 for validation, so it kind of makes sense imho

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -1350,6 +1350,24 @@ function custom_configuration()
             if [ -n "$networkingplugin" ] ; then
                 proposal_set_value neutron default "['attributes']['neutron']['networking_plugin']" "'$networkingplugin'"
             fi
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: I think you need to put the 6 lines above in a `else`.

Comment (toabctl) [inline scripts/qa_crowbarsetup.sh]: > I think you need to put the 6 lines above in a else.

Done in https://github.com/toabctl/automation/commit/69a064c3fe4e5cc592a7f4e6e4dcc9e42d22ef9a

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -1344,12 +1344,32 @@ function custom_configuration()
...
-            if [ -n "$networkingmode" ] ; then
-                proposal_set_value neutron default "['attributes']['neutron']['networking_mode']" "'$networkingmode'"
-            fi
-            if [ -n "$networkingplugin" ] ; then
-                proposal_set_value neutron default "['attributes']['neutron']['networking_plugin']" "'$networkingplugin'"
+
+            # For Cloud > 5 M4, proposal attribute names changed
+            # TODO(toabctl): the milestone/cloud6 check can be removed when milestone 5 is released
+            if iscloudver 5plus && [[ ! $cloudsource =~ ^M[1-4]+$ ]] || iscloudver 6plus; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: iscloudver 5 && ...

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -1344,12 +1344,32 @@ function custom_configuration()
...
+                if [ "$networkingplugin" = openvswitch ] ; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: please also quote the right side of the comparison for consistency

**Comment by dirkmueller** [comment]
+1

**Pull request merged**